### DISCUSSION TITLE
Automatically remove any .tmp files when starting up a project.

### DISCUSSION THREAD

**Discussion by nathnolt** [godotengine/godot-proposals#8164] [Editor]
Godot (at least 4.2 beta) 
Seems to be creating .tmp files in my project. Some of these .tmp files are days old. The .tmp extension is inside of the .gitignore so these are definitely is not needed and can be removed safely (at least whilst Godot is not running).
It kind of clutters up the file list, having more tmp files than actually useful files.
This is kind of annoying when working with the file explorer or with third party code editors. It would be nice if Godot would auto remove these files either on startup of the project, or on closal.

I'm not sure if it's on every OS or just Windows though.

Thanks for reading

**Comment by nathnolt** [comment]
Alternatively / additionally, maybe Godot could make a dedicated tmp ( or .tmp ) folder inside of the root project directory instead of putting these files in a relevant folder amongst relevant files.

**Comment by AThousandShips** [comment]
Are you sure these are generated by the engine?

**Reply by nathnolt** [reply]
Actually, now that you say it, I'm not sure if Godot is actually doing this. It could also be Visual Studio Code or Github Desktop / another program doing this.

**Reply by AThousandShips** [reply]
I can't find any place except windows export that generates `.tmp` files

**Comment by Mickeon** [comment]
I work with 4.1 and random .tmp files may remain so frequently on at least one project, I had to tell git to ignore them. They were also the cause of an annoying bug it took me a while to fix, under the assumption that a directory only contained files I expected. So what's being suggested sounds nice, yes.

**Comment by AThousandShips** [comment]
If the files are generated by Godot then they should be removed correctly, otherwise Godot shouldn't interfere with other files generated, that would be harmful

**Comment by nathnolt** [comment]
Okay, so if this were to be worked on, then first it has to be clear why Godot even creates these .tmp files, and if this is strictly nessesary, and if it is nessesary, then Godot should track which files it created, and supposedly should store this in project settings, just so it can then remove them? Or maybe it could then instead put them into a tmp folder alongside the godot executable, or maybe in %appdata% ? just so it knows for sure that these files can be removed, rather than some other program creating tmp files into the project directory.

**Reply by AThousandShips** [reply]
Well the first question is: Does Godot generate these files? Because AFAIK it doesn't, unless it generates the suffix from somewhere else there's no use of that specific suffix anywhere in the code (except for windows export, but that's not what could be causing this)

**Reply by Mickeon** [reply]
Yes. By default it does.
![image](https://github.com/godotengine/godot-proposals/assets/66727710/b25abb21-debd-40e7-add5-8b1718630533)

All that does is set the FileAccess' `backup_save` to `true`. This is an internal thing that is not exposed to the user other than this setting.

The exact operation depends on the system. Sometimes these files are straight up not deleted as they should. Troubleshooting the cause is not exactly easy, because it happens sporadically even when the editor used completely normally.

**Reply by AThousandShips** [reply]
But those are not suffixed with "tmp", unless windows does that automaticallt behind the scene

I have none of these files in any of my projects, and I'm on windows, so it's not just a Godot + windows thing

**Reply by Mickeon** [reply]
🤷 It's beyond me now. This is the "closest" culprit I could find in the codebase.

**Reply by AThousandShips** [reply]
It does seem that windows does use that format, so the culprit here is that Godot fails to delete these, this is therefore a duplicate of:
* https://github.com/godotengine/godot/issues/82270

**Comment by AThousandShips** [comment]
This appears to be the same as:
* https://github.com/godotengine/godot/issues/82270

Please try on 4.2 and see as the bug should be solved there

**Comment by nathnolt** [comment]
Yeah, I went to 4.2 a couple of days ago, and so far, no tmp files. So I think it just was a bug or something?

**Reply by AThousandShips** [reply]
The bug above indeed, which was solved by https://github.com/godotengine/godot/pull/81001

**Reply by nathnolt** [reply]
Good digging. I will close this proposal.
